lib/foreman_maintain/utils/sync_plans.rb

```diff
@@ -0,0 +1,83 @@
+module ForemanMaintain
+  module Utils
+    module SyncPlans
+      include ForemanMaintain::Concerns::SystemHelpers
+
+      MAKE_SYNC_PLANS_DISABLE = 'add'.freeze
+      MAKE_SYNC_PLANS_ENABLE = 'remove'.freeze
+
+      class << self
+        include ForemanMaintain::Concerns::Hammer
+        include ForemanMaintain::Concerns::DataYamlHelpers
+        def disabled_plans_count
+          record_ids_needs_to_enable.length
+        end
+
+        def record_ids_needs_to_enable
+          existing_data = fetch_stored_data
+          return [] if update_infokeys_absent?(existing_data)
+          existing_data['sync_plans']['update']['ids'].to_a
+        end
+
+        def make_disable(ids)
+          update_records(ids, false)
+        end
+
+        def make_enable
+          ids = record_ids_needs_to_enable
+          update_records(ids, true, MAKE_SYNC_PLANS_ENABLE)
+        end
+
+        private
+
+        def update_records(ids, enabled, sync_plans_action = MAKE_SYNC_PLANS_DISABLE)
+          updated_record_ids = []
+          ids.each do |sp_id|
+            result = hammer("sync-plan update --id #{sp_id} --enabled #{enabled}")
+            if result.include?('Sync plan updated')
+              updated_record_ids << sp_id
+            else
+              raise result
+            end
+          end
+        ensure
+          save_to_yaml(updated_record_ids, sync_plans_action)
+        end
+
+        def save_to_yaml(ids, action_name)
+          data = sync_plans_update_info(fetch_stored_data)
+          data['sync_plans']['update']['ids'] = perform_yaml_action(
+            data['sync_plans']['update']['ids'], ids, action_name
+          )
+          store_data(data)
+        end
+
+        def perform_yaml_action(old_ids, new_ids, action_name)
+          old_ids ||= []
+          case action_name
+          when MAKE_SYNC_PLANS_DISABLE
+            old_ids.concat(new_ids)
+          when MAKE_SYNC_PLANS_ENABLE
+            old_ids -= new_ids
+          end
+          old_ids.uniq!
+          old_ids
```

Comment:
I guess you can remove the returning line of code and replace
`.uniq! with .uniq` unless you want to return `nil` for already uniq collection. 
 https://ruby-doc.org/core-2.2.0/Array.html#method-i-uniq-21